text/0000-compiler-team-trusted-contrib-maintainer.md

```diff
@@ -0,0 +1,381 @@
+
+As the team gets larger, our processes need to remain efficient. Final comment
+periods (FCPs) have traditionally required sign-off from all team members,
+which can become onerous with more team members. As the number of compiler team
+members has grown from ~10 to ~15 since RFC 2689, the team has already noticed
+scaling issues with our FCP process.
+
+Processes which scale poorly with team size have acted as a unconscious
+disincentive to promote compiler team contributors to compiler team members.
+Similarly, the team has found that nominations being the primary mechanism for
+promotion to compiler team contributor or member tends to result in contributors
+falling through the cracks and being considered team members in the minds of the
+team but not actually having been nominated for promotion.
+
+Since RFC 2689, the compiler team contributor role's purpose has become
+confused. It is often beneficial to be able to grant the infrastructure
+and merge permissions to trusted contributors quickly so they can work more
+efficiently. However, it is also desirable for the compiler team contributor
+role to act as recognition for those contributors who have shown staying power
+and that the team would like to recognise. These goals are in tension, adding
+new contributors early and regularly improves the efficiency of the compiler
+team while watering down the recognition and sense of achievement that the role
+would ideally confer.
+
+In addition, as compiler team contributors and members increasingly leverage
+their contributions to gain employment/contracts to contribute to the project
+full-time (or otherwise), the naming of the compiler team contributor role can
+be confusing. An employer unfamiliar with the project may not realise that a
+compiler team contributor is a role within the project which recognises regular
+contribution and trust rather than just having made a handful of contributions
+and thus being a contributor.
+
+[rfc2689]: https://rust-lang.github.io/rfcs/2689-compiler-team-contributors.html
+[review_queue_analysis]: https://borrowed.dev/p/on-the-compiler-teams-review-queue
+
+Definitions
+-----------
+[definitions]: #definitions
+
+There are various permissions/privileges/responsibilities which will be
+referenced in later sections of this RFC, defined here:
+
+- **r+**
+  - Contributors with *r+* privilege are able to approve pull requests to be
+    merged by *bors*. Contributors should not merge their own pull requests (with
+    the exception of re-approving their own work on behalf of another contributor
+    after a rebase or similarly trivial change). *r+* permissions apply to the
+    whole repository, but [it is expected][expectations] that contributors limit
+    themselves to only those parts of the *rust-lang/rust* repository that are under
+    the purview of the compiler team (unless granted *r+* from other teams too), and
+    for subsystems/pull requests that they are confident reviewing.
+- **try**
```

Comment:
question that I'm gonna answer myself: why don't just give everyone permissions for try builds? the impact of the other perms is obvious, but it's notable that try permissions are a security risk as try builds have access to secrets, and they're also hosted on static.rust-lang.org and therefore must never contain malicious code

Reply:
Added some wording to this effect, thanks!